### PR TITLE
Add player message for in_check? check_game_status in pieces_controller

### PR DESCRIPTION
**Pull request by megankbarker**
I've connected in_check? method in the pieces_controller.

### CODE REVIEW

File: app/controllers/pieces_controller.rb [hunk 1]
```diff
@@ -213,7 +213,16 @@ def check_game_status
 
     # Methods to call
     # Checkmate, lose or win
-    # Check
+    # Check with in_check? method
+    if game.in_check?(color)
```
Comment (woodbrettm) [inline app/controllers/pieces_controller.rb]: Should be @game

File: app/controllers/pieces_controller.rb [hunk 2]
```diff
@@ -213,7 +213,16 @@ def check_game_status
...
+      add_player_messages('Ugh... Your king is in check!', 'Nice! You\'ve put opponent\'s king in check!')
+      check_game_status
```
Comment (woodbrettm) [inline app/controllers/pieces_controller.rb:218]: don't need this

File: app/controllers/pieces_controller.rb [hunk 3]
```diff
@@ -213,7 +213,16 @@ def check_game_status
...
+      @game.update_player_turn
```
Comment (woodbrettm) [inline app/controllers/pieces_controller.rb:219]: Don't need this

File: app/controllers/pieces_controller.rb [hunk 4]
```diff
@@ -213,7 +213,16 @@ def check_game_status
...
+      200
```
Comment (woodbrettm) [inline app/controllers/pieces_controller.rb]: Don't need the 200 or 422 status returns here. def check_game_status is only to override the player messages. So really all you'll need is `if @game.in_check? add_player_messages(etc, etc) elsif #add next game check here end`

**Comment by megankbarker** [comment]
- updated the game.in_check? invocation to be @game.in_check?
- removed 200 and 422 references

**Comment by megankbarker** [comment]
Updated game_check_method

**Pull request merged**